bluesky_widgets/qt/search_input.py

```diff
@@ -16,54 +26,168 @@ def __init__(self, model, *args, **kwargs):
+                        "is timedelta() or None."
+                    )
+                for item in allowed:
+                    if since == item:
+                        break
+                else:
+                    # No matches
+                    raise ValueError(
+                        "This UI can only express since as a timedelta if it is "
+                        f"one of {allowed}. The value {since} is not allowed"
+                    )
+        s = SearchInput()
+        s.time_validator = time_validator
 
         # "Since: <datetime picker>"
         self.since_widget = QDateTimeEdit()
         self.since_widget.setCalendarPopup(True)
         self.since_widget.setDisplayFormat("yyyy-MM-dd HH:mm")
         self.layout().addRow("Since:", self.since_widget)
+        self.since_widget.dateTimeChanged.connect(self.uncheck_radiobuttons)
 
         # "Until: <datetime picker>"
         self.until_widget = QDateTimeEdit()
         self.until_widget.setCalendarPopup(True)
         self.until_widget.setDisplayFormat("yyyy-MM-dd HH:mm")
         self.layout().addRow("Until:", self.until_widget)
+        self.until_widget.dateTimeChanged.connect(self.uncheck_radiobuttons)
 
+        # Refresh Button
         self.refresh_button = QPushButton("Refresh")
         self.layout().addWidget(self.refresh_button)
 
         # Initialize values.
         qdatetime = QDateTime()
-        qdatetime.setSecsSinceEpoch(self.model.since)
+        qdatetime.setSecsSinceEpoch(self.model.since.timestamp())
         self.since_widget.setDateTime(qdatetime)
         qdatetime = QDateTime()
-        qdatetime.setSecsSinceEpoch(self.model.until)
+        qdatetime.setSecsSinceEpoch(self.model.until.timestamp())
         self.until_widget.setDateTime(qdatetime)
 
         # Changes to the GUI update the model.
         self.since_widget.dateTimeChanged.connect(self.on_since_view_changed)
         self.until_widget.dateTimeChanged.connect(self.on_until_view_changed)
-        self.refresh_button.clicked.connect(self.model.events.reload)
+
+        # TODO: check what refresh button is doing
+        self.refresh_button.clicked.connect(self.model.request_reload)
+        self.model.events.reload.connect(self.on_reload)
+        self.model.events.query.connect(self.on_reload)
         # Changes to the model update the GUI.
         self.model.events.since.connect(self.on_since_model_changed)
         self.model.events.until.connect(self.on_until_model_changed)
 
+        # connect QRadioButtons and change date dropdowns (since/until widgets) accordingly
+        self.today_widget.clicked.connect(self.on_select_today)
+        self.hour_widget.clicked.connect(self.on_select_lasthour)
+        self.days_widget.clicked.connect(self.on_select_30days)
+        self.all_widget.clicked.connect(self.on_select_all)
+
+    def on_reload(self, event):
+        now = time.time()
+        if isinstance(event.date, timedelta):
```

Comment:
As you know, this is connected to the `self.model.events.reload` signal and the `self.model.events.query` signal. Those signals event `Event`s that do not have a date attached. (They are effectively just a timing signal: "It happened now!" with no extra content.) So, instead of `event.date` we need to use `self.model.since` here.